src/CaptureFileInfo/ItemModel.cpp

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "CaptureFileInfo/ItemModel.h"
+
+#include "CaptureFileInfo/CaptureFileInfo.h"
+#include "CoreUtils.h"
+#include "OrbitBase/Logging.h"
+
+namespace orbit_capture_file_info {
+
+void ItemModel::SetCaptureFileInfos(std::vector<CaptureFileInfo> capture_file_infos) {
+  beginInsertRows(QModelIndex{}, 0, static_cast<int>(capture_file_infos.size()) - 1);
+  capture_files_ = std::move(capture_file_infos);
+  endInsertRows();
```

Comment:
You're replacing all entries in capture_files_ which means we have to express that as deleting all rows first and then adding new ones. So this needs some `beginRemoveRows` as well.

Reply:
This is also not taking into account that `capture_file_infos` could be empty.

Reply:
done